include/seqan3/alphabet/union_alphabet.hpp

```diff
@@ -43,225 +43,292 @@
  *     // doesn't work:
  *     // union_alphabet<dna4, gap> my_letter{'A'};
  *
  *     union_alphabet<dna4, gap>.assign_char('C'); // <- this does!
- *
  *     union_alphabet<dna4, gap>.assign_char('-'); // gap character
+ *     union_alphabet<dna4, gap>.assign_char('K'); // unknown characters map to the default/unknown
+ *                                               // character of the first alphabet type (i.e. A of dna4)
  *     if (my_letter.to_char() == 'A')
  *        std::cout << "yeah\n"; // "yeah";
+ * ```
  */
-
 template <typename first_alphabet_type, typename ...alphabet_types>
     requires alphabet_concept<first_alphabet_type> && (alphabet_concept<alphabet_types> && ...)
-struct union_alphabet
+class union_alphabet
 {
-    /* types */
 public:
-    //! The size of the alphabet, i.e. the number of different values it can take.
+    /* types */
+    //!\brief The size of the alphabet, i.e. the number of different values it can take.
     static constexpr size_t value_size = (alphabet_types::value_size + ... + first_alphabet_type::value_size);
-    //! the type of the alphabet when converted to char (e.g. via @link to_char @endlink)
+    //!\brief the type of the alphabet when converted to char (e.g. via \link to_char \endlink)
     using char_type = typename first_alphabet_type::char_type;
-    //! the type of the alphabet when represented as a number (e.g. via @link to_rank @endlink)
-    using rank_type = decltype(detail::at_least_type<value_size>());
-
-protected:
-    using tuple_t = std::tuple<first_alphabet_type, alphabet_types...>;
-    using index_sequence_t = decltype(std::make_index_sequence<std::tuple_size_v<tuple_t>>{});
-    static constexpr auto sizes = detail::alphabet_sizes<tuple_t>(index_sequence_t{});
-    static constexpr auto alphabets = detail::alphabet_objects<tuple_t>(index_sequence_t{});
+    //!\brief the type of the alphabet when represented as a number (e.g. via \link to_rank \endlink)
+    using rank_type = detail::min_viable_uint_t<value_size>;
+    /*!\brief the type used to assign a value from one of the base alphabets
+     * during copy construction or copy assignment.
+     *
+     * (i.e. via
+     * \link union_alphabet(const variant_type & alphabet) \endlink or
+     * \link operator=(const variant_type & alphabet) \endlink )
+     */
+    using variant_type = std::variant<first_alphabet_type, alphabet_types...>;
+
+    //!\name default constructors
+    //!\{
+    constexpr union_alphabet() = default;
+    constexpr union_alphabet(union_alphabet const &) = default;
+    constexpr union_alphabet(union_alphabet &&) = default;
+    //!\}
+
+    //!\name default assignment operators
+    //!\{
+    constexpr union_alphabet & operator= (union_alphabet const &) = default;
+    constexpr union_alphabet & operator= (union_alphabet &&) = default;
+    //!\}
+
+    //!\name default assignment operators
+    //!\{
+    explicit constexpr union_alphabet(rank_type && value)
+        : _value{value}
```

Comment:
I can remove this constructor all together. I rewrote all test cases, s.t. this constructor is not used anymore.